New HAL API for remap

asmorkalov opened the pull request:
Address https://github.com/opencv/opencv/issues/25392

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

opencv-alalek commented on modules/imgproc/src/imgwarp.cpp:
```diff
@@ -1816,6 +1815,10 @@ void cv::remap( InputArray _src, OutputArray _dst,
         src = src.clone();
 
+    if ((map1.type() == CV_32FC1) && (map2.type() == CV_32FC1))
+        CALL_HAL(remap, cv_hal_remap, src.type(), src.data, src.step, src.cols, src.rows, dst.data, dst.step, dst.cols, dst.rows,
```
In general HAL API should be called unconditionally.

Or we need to rename API and apply `32f` suffix.

asmorkalov commented on modules/imgproc/src/hal_replacement.hpp:
```diff
@@ -298,6 +298,36 @@ inline int hal_ni_warpPerspective(int src_type, const uchar *src_data, size_t sr
+inline int hal_ni_remap(int src_type, const uchar *src_data, size_t src_step, int src_width, int src_height,
+                        uchar *dst_data, size_t dst_step, int dst_width, int dst_height,
+                        float* mapx, size_t mapx_step, float* mapy, size_t mapy_step,
+                        int interpolation, int borderType, const double borderValue[4])
```
borderValue should be changed to const uchar* border_value and assume that caller (OpenCV) converted border_value to the proper type. - It's incompatible with cv::Scalar. It's polymorphic.

asmorkalov commented:
@opencv-alalek @vpisarev I fixed remarks:
- Renamed HAL function with 32f prefix.
- Stay with existing approach for border value like in other functions like warpAffine, warpPerspective.
